business_rules/utils.py

```diff
@@ -85,3 +86,86 @@ def params_dict_to_list(params):
             'field_type': param_field_type
         } for name, param_field_type in params.items()
     ]
+
+
+def validate_rule_data(variables, actions, rule):
+    """
+    validate_rule_data is used to check a generated rule against a set of variables and actions
+    :param variables:
+    :param actions:
+    :param rule:
+    :return: bool
+    :raises AssertionError:
+    """
+    def validate_root_keys(rule):
+        """
+        Check the root object contains both 'actions' & 'conditions'
+        """
+        root_keys = rule.keys()
+        if 'actions' not in root_keys:
+            raise AssertionError('Missing "{}" key'.format('actions'))
+        if 'conditions' not in root_keys:
+            raise AssertionError('Missing "{}" key'.format('conditions'))
+
+    def validate_condition_operator(condition, rule_schema):
+        """
+        Check provided condition contains a valid operator
+        """
+        for item in rule_schema.get('variables'):
+            if item.get('name') == condition.get('name'):
+                condition_field_type = item.get('field_type')
+                variable_operators = rule_schema.get('variable_type_operators', {}).get(condition_field_type, [])
+                for operators in variable_operators:
+                    if operators['name'] == condition['operator']:
+                        return True
+                raise AssertionError('Unknown operator "{}"'.format(condition['operator']))
+
+    def validate_condition_name(condition, variables):
+        """
+        Check provided condition contains a 'name' key and the value is valid
+        """
+        condition_name = condition.get('name')
+        if not condition_name:
+            raise AssertionError('Missing condition "name" key in {}'.format(condition))
+        if not hasattr(variables, condition_name):
+            raise AssertionError('Unknown condition "{}"'.format(condition_name))
+
+    def validate_condition(condition, variables, rule_schema):
+        validate_condition_name(condition, variables)
+        validate_condition_operator(condition, rule_schema)
+        method = getattr(variables, condition.get('name'))
+        params = condition.get('params', {})
+        engine._check_params_valid_for_method(method, params, method_type.METHOD_TYPE_VARIABLE)
```

Comment:
I think it makes more sense to move this function into this module and reuse it from `engine` module.